cluster/saltbase/salt/kube-addons/initd

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+#
+### BEGIN INIT INFO
+# Provides:   kube-addons
+# Required-Start:    $local_fs $network $syslog kube-apiserver
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Kubernetes Addon Object Manager
+# Description:
+#   Enforces installation of Kubernetes Addon Objects
+### END INIT INFO
+
+
+# PATH should only include /usr/* if it runs after the mountnfs.sh script
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+DESC="Kubernetes Addon Object Manager"
+NAME=kube-addons
+DAEMON_LOG_FILE=/var/log/$NAME.log
+PIDFILE=/var/run/$NAME.pid
+SCRIPTNAME=/etc/init.d/$NAME
+KUBECTL=/usr/local/bin/kubectl
+
+# Define LSB log_* functions.
+# Depend on lsb-base (>= 3.2-14) to ensure that this file is present
+# and status_of_proc is working.
+. /lib/lsb/init-functions
+
+function addon_manager_async() {
```

Comment:
I was worried about that, but the parallel spray works fine (at the
moment). In the future I could imagine a rudimentary approach using tsort.

On Tue, Jan 20, 2015 at 10:39 AM, Eric Tune <notifications@github.com>
wrote:

> In cluster/saltbase/salt/kube-addons/initd
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3601#discussion-diff-23243941>
> :
>
> > +
> > +# PATH should only include /usr/* if it runs after the mountnfs.sh script
> > +PATH=/sbin:/usr/sbin:/bin:/usr/bin
> > +DESC="Kubernetes Addon Object Manager"
> > +NAME=kube-addons
> > +DAEMON_LOG_FILE=/var/log/$NAME.log
> > +PIDFILE=/var/run/$NAME.pid
> > +SCRIPTNAME=/etc/init.d/$NAME
> > +KUBECTL=/usr/local/bin/kubectl
> > +
> > +# Define LSB log_* functions.
> > +# Depend on lsb-base (>= 3.2-14) to ensure that this file is present
> > +# and status_of_proc is working.
> > +. /lib/lsb/init-functions
> > +
> > +function addon_manager_async() {
>
> Do we need some way to sequence the creation of these objects?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3601/files#r23243941>
> .
>